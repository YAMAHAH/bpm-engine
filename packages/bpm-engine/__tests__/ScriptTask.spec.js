import fs from 'fs';

import BPMEngine from 'bpm-engine';

import History from './Plugins/History';

describe('ScriptTask', () => {
  it('should run a script task successfullt', async () => {
    // given
    const history = new History();
    const bpm = new BPMEngine({
      plugins: [history],
    });

    const token = await bpm.createProcessInstance({
      workflowDefinition: fs.readFileSync(`${__dirname}/diagrams/ScriptTask.bpmn`, 'utf-8'),
    });

    // when
    await token.execute();

    // // then
    expect(token.payload).toMatchSnapshot();
    expect(token.status).toBe('ended');
  });

  it('should break when using a unsupported script format', async () => {
    // given
    const history = new History();
    const bpm = new BPMEngine({
      plugins: [history],
    });

    const token = await bpm.createProcessInstance({
      workflowDefinition: fs.readFileSync(
        `${__dirname}/diagrams/ScriptTaskWithUnsupportedScriptFormat.bpmn`,
        'utf-8',
      ),
    });

    // when
    await expect(token.execute()).rejects.toMatchSnapshot();

    // // then
    expect(token.status).toBe('paused');
  });
});
