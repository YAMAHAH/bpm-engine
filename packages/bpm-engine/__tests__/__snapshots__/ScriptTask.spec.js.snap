// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScriptTask should break when using a unsupported script format 1`] = `[Error: Unsupported script format: PHP. Allowed script formats: JavaScript, NodeJS]`;

exports[`ScriptTask should run a script task successfullt 1`] = `
Object {
  "a": 1,
}
`;
